tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -61,31 +59,30 @@ public static AIBotConfig fromMailetConfig(MailetConfig mailetConfig) throws Mai
         if (Strings.isNullOrEmpty(gptAddressParam)) {
             throw new MailetException("No value for " + GPT_ADDRESS_PARAMETER_NAME + " parameter was provided.");
         }
-        
-        MailAddress mailAddress;
+
+        Optional<URL> baseURLOpt = toOptionalIfNotEmpty(baseUrlParam)
+                .flatMap(AIBotConfig::baseURLStringToURL);
+
+        LlmModel llmModel = !Strings.isNullOrEmpty(llmModelParam)
+                            ? parseLlmModelParamOrThrow(llmModelParam)
+                            : DEFAULT_LLM_MODEL;
+
         try {
-            mailAddress = new MailAddress(gptAddressParam);
-        } catch (AddressException e) {
+            return new AIBotConfig(apiKeyParam, new MailAddress(gptAddressParam), llmModel, baseURLOpt.orElse(null));
+        } catch (Exception e) {
             throw new RuntimeException(e);
         }
 
-        LlmModel llmModel;
-        if (Strings.isNullOrEmpty(llmModelParam)) {
-            llmModel = DEFAULT_LLM_MODEL;
-        } else {
-            llmModel = parseLlmModelParamOrThrow(llmModelParam);
-        }
+    }
 
-        URL baseURL = null;
+    private static @NotNull Optional<URL> baseURLStringToURL(String baseUrlString) {
         try {
-            if (!Strings.isNullOrEmpty(baseUrlParam)) {
-                baseURL = URI.create(baseUrlParam).toURL();
-            }
+            return Optional.of(URI.create(baseUrlString).toURL());
         } catch (MalformedURLException e) {
             LOGGER.warn("Invalid LLM API base URL", e);
         }
 
-        return new AIBotConfig(apiKeyParam, baseURL, mailAddress, llmModel);
+        return Optional.empty();
```

Comment:
We are accepting invalid URL, ignoring them and just start with the default one.

Good chance the user do not notice easily what's wrong.

Aborting the configuration sequence in case of bad values is perfectly Ok.

IMO we should propagate this exception.